tfsdk/attribute.go

```diff
@@ -341,18 +341,34 @@ func (a Attribute) validate(ctx context.Context, req ValidateAttributeRequest, r
 				}
 			}
 		case NestingModeSingle:
-			for nestedName, nestedAttr := range a.Attributes.GetAttributes() {
-				nestedAttrReq := ValidateAttributeRequest{
-					AttributePath: req.AttributePath.WithAttributeName(nestedName),
-					Config:        req.Config,
-				}
-				nestedAttrResp := &ValidateAttributeResponse{
-					Diagnostics: resp.Diagnostics,
-				}
+			o, ok := req.AttributeConfig.(types.Object)
 
-				nestedAttr.validate(ctx, nestedAttrReq, nestedAttrResp)
+			if !ok {
+				err := fmt.Errorf("unknown attribute value type (%T) for nesting mode (%T) at path: %s", req.AttributeConfig, nm, req.AttributePath)
+				resp.Diagnostics = append(resp.Diagnostics, &tfprotov6.Diagnostic{
```

Comment:
Really sorry to do this to you, @ewbankkit, but could you update to use the helpers added by #110? Just want to keep our consistency @bflad worked so hard to introduce. 😦 

Sorry, ping me in chat when you do that and we'll get it merged straightaway before something else comes up. Thanks so much for the patience on this.